ledger-service/http-json/src/itlib/scala/http/FilterDiscriminatorScenario.scala

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.http
+
+import com.daml.lf.value.test.TypedValueGenerators.{ValueAddend => VA}
+import spray.json.JsValue
+
+/** A query, a value that matches the query, and a value that doesn't match.
+  */
+class FilterDiscriminatorScenario[Inj](
+    val label: String,
+    val ctId: domain.ContractTypeId.OptionalPkg,
+    val va: VA.Aux[Inj],
+    val query: Map[String, JsValue],
+    val matches: domain.Party => Inj,
+    val doesNotMatch: domain.Party => Inj,
+)
```

Comment:
This type can also form the basis of Scalacheck-style checking of payload queries.